scionlab/tests/test_user_as_models.py

```diff
@@ -213,97 +248,132 @@ def _check_attachment_point(testcase, attachment_point):
+            att_conf_dict['public_ip'] = public_ip
+        else:
+            att_conf_dict['public_ip'] = None
+        att_conf_dict['public_port'] = public_port
+
+        while True:
+            bind_ip = '192.168.1.%i' % r.randint(10, 254)
+            bind_port = r.choice(range(DEFAULT_PUBLIC_PORT + 1000, DEFAULT_PUBLIC_PORT + 1020))
+            if (bind_ip, bind_port) not in used_bind_ip_port_pairs:
+                used_bind_ip_port_pairs.add((bind_ip, bind_port))
+                break
+        if _randbool(r) or force_bind_ip:
+            att_conf_dict['bind_ip'] = bind_ip
+            att_conf_dict['bind_port'] = bind_port
+        else:
+            att_conf_dict['bind_ip'] = None
+            att_conf_dict['bind_port'] = None
+        att_confs.append(AttachmentConf(**att_conf_dict))
+
+    return att_confs
+
+
+def _get_random_useras_params(seed, vpn_choice, **kwargs):
     """
     Generate some "random" parameters for a UserAS based on `seed`.
     Any parameters to UserAS.objects.create can be specified to override the generated values.
-    Note: overriding parameters may affect the generated value for other parameters.
     :returns: kwargs dict for UserAS.objects.create
     """
     r = random.Random(seed)
 
-    def _randbool():
-        return r.choice([True, False])
-
-    use_vpn = kwargs.setdefault('use_vpn', _randbool())
-    # Make AP choice both with/without VPN to always consume same amount of random state
-    candidate_AP_vpn = r.choice(AttachmentPoint.objects.filter(vpn__isnull=False))
-    candidate_AP_novpn = r.choice(AttachmentPoint.objects.all())
-    if use_vpn:
-        kwargs.setdefault('attachment_point', candidate_AP_vpn)
-    else:
-        kwargs.setdefault('attachment_point', candidate_AP_novpn)
     kwargs.setdefault('owner', get_testuser())
-
-    public_ip = '172.31.0.%i' % r.randint(10, 254)
-    public_port = r.choice(range(DEFAULT_PUBLIC_PORT, DEFAULT_PUBLIC_PORT + 20))
-    if _randbool() or not use_vpn or force_public_ip:
-        kwargs.setdefault('public_ip', public_ip)
-    else:
-        kwargs.setdefault('public_ip', None)
-    kwargs.setdefault('public_port', public_port)
-
-    bind_ip = '192.168.1.%i' % r.randint(10, 254)
-    bind_port = r.choice(range(DEFAULT_PUBLIC_PORT + 1000, DEFAULT_PUBLIC_PORT + 1020))
-    if _randbool() or force_bind_ip:
-        kwargs.setdefault('bind_ip', bind_ip)
-        kwargs.setdefault('bind_port', bind_port)
-    else:
-        kwargs.setdefault('bind_ip', None)
-        kwargs.setdefault('bind_port', None)
-
     kwargs.setdefault('installation_type', r.choice((UserAS.VM, UserAS.PKG, UserAS.SRC)))
     randstr = r.getrandbits(1024).to_bytes(1024//8, 'little').decode('utf8', 'ignore')
     kwargs.setdefault('label', randstr)
 
     return kwargs
 
 
-def create_random_useras(testcase, seed, **kwargs):
+def create_and_check_random_useras(testcase, seed, ases_defs, vpn_choice, **kwargs):
```

Comment:
True, but this way we would not parameterize on the attachment points combinations, hence the tests instead of being executed once for each attachment points combination, it would get executed only once.
Sure this is feasible, but the tests would be less comprehensive